redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/protocolHostGeneration.kt

```diff
@@ -371,30 +377,25 @@ internal fun generateProtocolNode(
                         KotlinxSerialization.jsonBoolean,
                       )
                       val arguments = mutableListOf<CodeBlock>()
-                      for ((index, parameterFqType) in trait.parameterTypes.withIndex()) {
+                      for (parameterFqType in trait.parameterTypes) {
                         val parameterType = parameterFqType.asTypeName()
                         val serializerId = serializerIds.computeIfAbsent(parameterType) {
                           nextSerializerId++
                         }
-                        arguments += CodeBlock.of(
-                          "json.encodeToJsonElement(serializer_%L, arg%L)",
-                          serializerId,
-                          index,
-                        )
+                        arguments += CodeBlock.of("serializer_%L", serializerId)
                       }
                       if (trait.parameterTypes.isEmpty()) {
-                        beginControlFlow("{")
+                        addStatement(
+                          "%L(json, change.id, eventSink)::invoke",
```

Comment:
Uuuuuggghhh I hate it. So wasteful. I wonder if ES6 or K2 or both will remove that implementation limitation for future optimization.

Reply:
Same.

Reply:
https://youtrack.jetbrains.com/issue/KT-29580/JS-allow-to-inherit-from-Function-interfaces-in-JS-IR-BE